scalardb/src/scalardb/transfer.clj

```diff
@@ -130,8 +130,14 @@
   "Read a record with a transaction. If read fails, this function returns nil."
   [tx storage i]
   (try
-    (.get tx (prepare-get i))
-    (.get storage (prepare-get i))
+    ;; Need Storage API to read the transaction metadata
+    (let [tx-result (.get tx (prepare-get i))
+          result (.get storage (prepare-get i))]
+      ;; Put the same balance to check conflicts with in-flight transactions
+      (->> (calc-new-balance tx-result 0)
+           (prepare-put i)
+           (.put tx))
+      result)
```

Comment:
I think I missed something, but I thought the transactional Get can also detect conflicts with in-flight transactions (by throwing `UncommittedRecordException`). Why do we need this additional dummy write?

Reply:
Read skew could happen because this is a read-only transaction when the isolation level is snapshot isolation.

Reply:
Understood. So it's like a manual `EXTRA_WRITE`. Thanks.